internal/api/migrate/loader.go

```diff
@@ -14,112 +14,89 @@
+	var result []*build.Instance
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return nil, err
 	}
-	loaders := service.GetLoaders()
-
-	return &schemas.Watcher{
-			FSWatcher:      fsWatcher,
-			Loaders:        loaders,
-			LoaderCallback: callback,
-		}, &schemas.Reloader{
-			Loaders:        loaders,
-			LoaderCallback: callback,
-		},
-		nil
-}
-
-type loader interface {
-	schemas.Loader
-	getConditions() string
-	getPlaceholder() string
-}
-
-type migCuePart struct {
-	listOfConditions atomic.Pointer[string]
-	schemasPath      string
-	defaultValue     string
-	placeholderText  string
-}
-
-func (c *migCuePart) GetSchemaPath() string {
-	return c.schemasPath
-}
-
-// Load will load the migration schemas from the filesystem
-func (c *migCuePart) Load() (int, int, error) {
-	var conditions string
-	conditions, successfulLoadsCount, err := c.buildListOfConditions()
-	if err == nil {
-		c.listOfConditions.Store(&conditions)
+	for _, file := range files {
+		if !file.IsDir() {
+			logrus.Tracef("file %s is ignored since we are looking for directories", file.Name())
+			continue
+		}
+		schemaPath := filepath.Join(path, file.Name(), "migrate")
+		if _, fileErr := os.Stat(filepath.Join(schemaPath, "migrate.cue")); os.IsNotExist(fileErr) {
+			// migration file doesn't exist
+			continue
+		}
+		buildInstance, loadErr := schemas.LoadMigrateSchema(schemaPath)
+		if loadErr != nil {
+			return nil, loadErr
+		}
+		result = append(result, buildInstance)
 	}
-	// NB: There are no relevant case of "failed loads" to monitor in the case of migration schemas
-	// at the moment, so we always return 0 for the failed loads count
-	return successfulLoadsCount, 0, err
-}
-
-func (c *migCuePart) getConditions() string {
-	return *c.listOfConditions.Load()
-}
-
-func (c *migCuePart) getPlaceholder() string {
-	return c.placeholderText
+	return result, nil
 }
 
-func (c *migCuePart) buildListOfConditions() (string, int, error) {
-	successfulLoadsCount := 0
-
-	files, err := os.ReadDir(c.schemasPath)
+func loadPanels(panelSchemaPath string) (map[string]*build.Instance, error) {
+	ctx := cuecontext.New(cuecontext.EvaluatorVersion(cuecontext.EvalV3))
+	result := make(map[string]*build.Instance)
+	files, err := os.ReadDir(panelSchemaPath)
 	if err != nil {
-		return "", successfulLoadsCount, err
+		return nil, err
 	}
-
-	// gather the content of all migration files found
-	var listOfConditions strings.Builder
 	for _, file := range files {
 		if !file.IsDir() {
 			logrus.Tracef("file %s is ignored since we are looking for directories", file.Name())
 			continue
 		}
-		migFilePath := filepath.Join(c.schemasPath, file.Name(), "migrate.cue")
-		contentStr, readErr := os.ReadFile(migFilePath)
-		if readErr != nil {
-			logrus.WithError(readErr).Debugf("No migration file found at %s, plugin %s will be skipped", migFilePath, file.Name())
+		schemaPath := filepath.Join(panelSchemaPath, file.Name(), "migrate")
```

Comment:
to be adjusted with the new `migrate` folder added

Reply:
that's already the case ;)